### PR TITLE
The Poincare polydisk product metric.

### PR DESCRIPTION
**Pull request by YannCabanes**
Define the Poincare polydisk product metric.

### CODE REVIEW

File: tests/test_hyperbolic_space.py [hunk 1]
```diff
@@ -438,3 +438,44 @@ def test_mean_and_belongs(self):
...
+    @geomstats.tests.np_only
+    def test_scaled_inner_product(self):
+        base_point_intrinsic = gs.array([1, 1, 1])
+        base_point = self.space.intrinsic_to_extrinsic_coords(base_point_intrinsic)
+        tangent_vec_a = gs.array([1, 2, 3, 4])
+        tangent_vec_b = gs.array([5, 6, 7, 8])
+        tangent_vec_a = self.space.projection_to_tangent_space(tangent_vec_a, base_point)
+        tangent_vec_b = self.space.projection_to_tangent_space(tangent_vec_b, base_point)
+        scale = 2
+        default_metric = HyperbolicSpace(dimension=self.dimension)
```
Comment (ninamiolane) [inline tests/test_hyperbolic_space.py]: should be `default_space`?

File: tests/test_hyperbolic_space.py [hunk 2]
```diff
@@ -438,3 +438,44 @@ def test_mean_and_belongs(self):
...
+        scaled_metric = HyperbolicSpace(dimension=self.dimension, scale=2)
```
Comment (ninamiolane) [inline tests/test_hyperbolic_space.py]: should be `scaled_space`

File: tests/test_hyperbolic_space.py [hunk 3]
```diff
@@ -438,3 +438,44 @@ def test_mean_and_belongs(self):
...
+        self.assertAllClose(scale ** 2 * inner_product_default_metric, inner_product_scaled_metric)
+
+    @geomstats.tests.np_only
+    def test_scaled_squared_norm(self):
+        base_point_intrinsic = gs.array([1, 1, 1])
+        base_point = self.space.intrinsic_to_extrinsic_coords(base_point_intrinsic)
+        tangent_vec = gs.array([1, 2, 3, 4])
+        tangent_vec = self.space.projection_to_tangent_space(tangent_vec, base_point)
+        scale = 2
+        default_metric = HyperbolicSpace(dimension=self.dimension)
```
Comment (ninamiolane) [inline tests/test_hyperbolic_space.py]: same

File: tests/test_hyperbolic_space.py [hunk 4]
```diff
@@ -438,3 +438,44 @@ def test_mean_and_belongs(self):
...
+        scaled_metric = HyperbolicSpace(dimension=self.dimension, scale=2)
```
Comment (ninamiolane) [inline tests/test_hyperbolic_space.py]: same

File: tests/test_hyperbolic_space.py [hunk 5]
```diff
@@ -438,3 +438,44 @@ def test_mean_and_belongs(self):
...
+        self.assertAllClose(scale ** 2 * squared_norm_default_metric, squared_norm_scaled_metric)
+
+    @geomstats.tests.np_only
+    def test_scaled_distance(self):
+        point_a_intrinsic = gs.array([1, 2, 3])
+        point_b_intrinsic = gs.array([4, 5, 6])
+        point_a = self.space.intrinsic_to_extrinsic_coords(point_a_intrinsic)
+        point_b = self.space.intrinsic_to_extrinsic_coords(point_b_intrinsic)
+        scale = 2
+        default_metric = HyperbolicSpace(dimension=self.dimension)
```
Comment (ninamiolane) [inline tests/test_hyperbolic_space.py]: same

File: tests/test_hyperbolic_space.py [hunk 6]
```diff
@@ -438,3 +438,44 @@ def test_mean_and_belongs(self):
...
+        scaled_metric = HyperbolicSpace(dimension=self.dimension, scale=2)
```
Comment (ninamiolane) [inline tests/test_hyperbolic_space.py]: same

File: tests/test_poincare_polydisk.py [hunk 7]
```diff
@@ -25,3 +26,23 @@ def test_metric_signature(self):
         expected = (self.n_disks * 2, 0, 0)
         result = self.metric.signature
         self.assertAllClose(result, expected)
+
+    def test_product_distance(self):
+        point_a_intrinsic = gs.array([0.01, 0.0])
+        point_b_intrinsic = gs.array([0.0, 0.0])
+        hyperbolic_space_instance = HyperbolicSpace(dimension=2)
```
Comment (ninamiolane) [inline tests/test_poincare_polydisk.py]: hyperbolic_space only

File: tests/test_poincare_polydisk.py [hunk 8]
```diff
@@ -25,3 +26,23 @@ def test_metric_signature(self):
...
+        point_a = hyperbolic_space_instance.intrinsic_to_extrinsic_coords(point_a_intrinsic)
+        point_b = hyperbolic_space_instance.intrinsic_to_extrinsic_coords(point_b_intrinsic)
+        list_points_shape = list(point_a.shape)
+        list_dimension_two_disks_data = [2] + list_points_shape
```
Comment (ninamiolane) [inline tests/test_poincare_polydisk.py]: same

Comment (ninamiolane) [reply tests/test_poincare_polydisk.py]: (2,) + point_a.shape

File: tests/test_hyperbolic_space.py (hunk 1 above)
Comment (YannCabanes) [reply tests/test_hyperbolic_space.py]: Resolved

File: tests/test_hyperbolic_space.py (hunk 2 above)
Comment (YannCabanes) [reply tests/test_hyperbolic_space.py]: Resolved

File: tests/test_hyperbolic_space.py (hunk 3 above)
Comment (YannCabanes) [reply tests/test_hyperbolic_space.py]: Resolved

File: tests/test_hyperbolic_space.py (hunk 4 above)
Comment (YannCabanes) [reply tests/test_hyperbolic_space.py]: Resolved

**Pull request merged**